CONTRIBUTING.md

```diff
@@ -0,0 +1,140 @@
+# Contributing Guidelines
+
+Thanks for thinking about contributing to Project Calico! The success of an
+open source project is entirely down to the efforts of its contributors, so we
+do genuinely want to thank you for contributing.
+
+This document contains some guidance and steps for contributing. Make sure you
+read it thoroughly, because if you miss some of these steps we will have to ask
```

Comment:
not convinced about this point - I think raising issues is certainly a key part of contributing and I think the contacting us is a key tool for assisting (which is how it's pitched here).  This isn't to replace our other contact info - but it's nice to have everything here rather than referencing a ton of different docs.